### PR TITLE
FireflowerS3Target: Fix format=Nop with local_s3_path (e.g. monsoon passthru compression)

### PR DESCRIPTION
**Pull request by jdanbrown**
- s3 path handling was correctly handling `format=Nop` by delegating to
  `S3Format.open`, but local path handling was incorrectly assuming that
  `compressed=False` implied "please decode as utf8", which was breaking
  in our dwellings monsoon use case where we passthru .zip bytes so the
  caller can deal with compression
- Also simplifies FireflowerS3Target to handle s3 vs. local more
  uniformly by delegating to the existing luigi S3Target and LocalTarget
  targets, which already understand `format`

Test plan:
- Added unit test

### CODE REVIEW

**Comment by kaloramik** [comment]
A lot of things use the compressed flag in signals and dwellings, I would probably check that it still works there. I'm not sure if the .open will work as the callers expect

**Comment by jdanbrown** [comment]
Oh duh, thanks.

**Comment by jdanbrown** [comment]
@kaloramik Actually, based on a grep of fireflower, signals, and dwellings, the only client of `FireflowerS3Target` that sets `.compressed` is `S3CSVTarget`:
- https://github.com/opendoor-labs/fireflower/blob/a97d900/fireflower/targets.py#L186
```sh
$ ack -w '\.compressed' ~/opendoor/{fireflower,signals,dwellings} --py --ignore-dir=venv
/Users/danb/opendoor/fireflower/fireflower/targets.py
231:        self.compressed = compressed
248:        if self.compressed:
262:            if self.compressed:
275:            if self.compressed:
295:            if self.compressed:
305:        if self.compressed:
323:            if self.compressed:
351:            if self.compressed:
363:            if self.compressed:

/Users/danb/opendoor/dwellings/dwellings/utils/utils.py
197:    if target.compressed:

$ ack -w 'setattr.*compressed' ~/opendoor/{fireflower,signals,dwel
lings} --py --ignore-dir=venv
# No results
```

Further, `S3CSVTarget` passes `format` in addition to setting `self.compressed`, and the change in this PR causes `format` to obviate `FireflowerS3Target`'s use of `.compressed` altogether:
- https://github.com/opendoor-labs/fireflower/blob/a97d900/fireflower/targets.py#L183-L189
```py
class S3CSVTarget(FireflowerS3Target):
    def __init__(self, path, compressed=True, kwargs_in=None, kwargs_out=None,
                 format=None, client=None):

        if compressed:
            format = luigi.format.Nop

        self.compressed = compressed
        self.kwargs_in = kwargs_in
        self.kwargs_out = kwargs_out
        super(S3CSVTarget, self).__init__(path, format, client)
```

I went ahead and added two more unit tests to encode this behavior and api:
- `test_compressed_is_compressed_FireflowerS3Target`
- `test_compressed_is_compressed_S3CSVTarget`

File: tests/targets_tests.py [hunk 1]
```diff
@@ -16,6 +18,66 @@
...
+            with mock.patch('fireflower.targets.os.getenv',
+                            side_effect=getenv):
+
+                # Passthru bytes with format=Nop instead of trying to encode/decode as utf8
+                s = FireflowerS3Target('some_file.txt.gz', format=luigi.format.Nop)
+                with s.open('w') as fout:
+                    fout.write(gzip.compress('asdf'.encode('utf8')))
+                    fout.write(gzip.compress('fdsa'.encode('utf8')))
+                with s.open('r') as fin:
+                    expected_bytes = fin.read()
```
Comment (kaloramik) [inline tests/targets_tests.py]: How come we want to get rid of compressed abilities in FireflowerS3Target? As in, now the user has to decompress on their own

Comment (jdanbrown) [inline tests/targets_tests.py]: Ah, no, we aren't getting rid of compression functionality:
- FireflowerS3Target wasn't compressing or decompressing in the first place. It was only encoding/decoding text (TextIOWrapper) or not.
- Now that we're passing through luigi's `format` in both the s3 and local case, we can get compression using luigi `format`'s directly: http://luigi.readthedocs.io/en/stable/_modules/luigi/format.html
```py
Text = TextFormat()
UTF8 = TextFormat(encoding='utf8')
Nop = NopFormat()
SysNewLine = NewlineFormat()
Gzip = GzipFormat()
Bzip2 = Bzip2Format()
MixedUnicodeBytes = MixedUnicodeBytesFormat()
```

**Comment by kaloramik** [comment]
LGTM

**Pull request merged**